check-trailing-whitespaces

```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+files_with_trailing_spaces=$(find . -type f -exec egrep -nH "\\s$" {} \; | sort -fh)
+
+if [[ ${files_with_trailing_spaces} ]]
+then
+    echo -e "\\e[97;41mTrailing whitespaces detected:\\e[0m"
+    e=$(printf '\033')
+    echo "${files_with_trailing_spaces}" | sed -E "s/^\\.\\/([^:]+):([0-9]+):(.*[^\t ])?([\t ]+)$/${e}[0;31m - in ${e}[0;33m\\1${e}[0;31m at line ${e}[0;33m\\2\\n   ${e}[0;31m>${e}[0m \\3${e}[41;1m\\4${e}[0m/"
```

Comment:
this `sed` is slightly different, in general it *looks* better (escaping), but I didn't run it to confirm, I assume tests are covering it ;)

Reply:
Yep they are supposed to :p